File: java/src/main/java/com/languageweaver/sdk/samples/translations/file/batch/EdgeBatchFileTranslationService.java
package com.languageweaver.sdk.samples.translations.file.batch;

import com.languageweaver.sdk.common.constants.PdfConverter;
import com.languageweaver.sdk.common.edge.EdgeLanguageWeaverClient;
import com.languageweaver.sdk.translate.edge.request.EdgeTranslateBatchFileRequest;

import java.io.IOException;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.ExecutionException;

public class EdgeBatchFileTranslationService {

    public static void main(String[] args) throws Exception {
        try (EdgeLanguageWeaverClient lwClient = new EdgeLanguageWeaverClient().build()) {
            translateBatchFile(lwClient);
            translateBatchFileWithLinguisticOptions(lwClient);
        }
    }

    private static void translateBatchFile(EdgeLanguageWeaverClient lwClient) throws ExecutionException, InterruptedException, IOException {
        EdgeTranslateBatchFileRequest translateFileRequest = new EdgeTranslateBatchFileRequest()
                .setLanguagePairId("EngFra_Generic_SRV_TNMV_8_5_x_1")
                // provide full path to the input and output folders
                .setInputBatchFolderPath(Paths.get("java", "src", "main", "resources", "input").toFile().getAbsolutePath())
                .setOutputBatchFolderPath(Paths.get("java", "src", "main", "resources", "output").toFile().getAbsolutePath())
                .setPdfConverter(PdfConverter.STANDARD)
                .addDictionary("DictionaryName1")
                .addDictionary("DictionaryName2");

        lwClient.translateBatchFile(translateFileRequest);
    }

    private static void translateBatchFileWithLinguisticOptions(EdgeLanguageWeaverClient lwClient) throws ExecutionException, InterruptedException, IOException {
        Map<String, String> linguisticOptions = new HashMap<>();
        linguisticOptions.put("formality", "Informal");
        EdgeTranslateBatchFileRequest translateFileRequest = new EdgeTranslateBatchFileRequest()
                .setLanguagePairId("EngJpn_Generic_SRV_TNMV_8_5_x_6")
                // provide full path to the input and output folders
                .setInputBatchFolderPath(Paths.get("java", "src", "main", "resources", "input").toFile().getAbsolutePath())
                .setOutputBatchFolderPath(Paths.get("java", "src", "main", "resources", "output").toFile().getAbsolutePath())
                .setLinguisticOptions(linguisticOptions)
                .addDictionary("DictionaryName1")
                .addDictionary("DictionaryName2");

        lwClient.translateBatchFile(translateFileRequest);
    }
}

File: dotnet/Samples/Translations/File/Batch/EdgeBatchFileTranslationService.cs
using LanguageWeaver.Sdk.Common.Edge;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.Translate.Edge.Request;

namespace LanguageWeaver.Sdk.Samples.Translations.File.Batch;

public class EdgeBatchFileTranslationService
{
    public static void Main()
    {
        using var lwClient = new EdgeLanguageWeaverClient().Build();

        TranslateBatchFile(lwClient);
        TranslateBatchFileWithLinguisticOptions(lwClient);
    }

    private static void TranslateBatchFile(EdgeLanguageWeaverClient lwClient)
    {
        var translateBatchFileRequest = new EdgeTranslateBatchFileRequest
        {
            LanguagePairId = "EngFra_Generic_SRV_TNMV_8_5_x_1",
            PdfConverter = PdfConverter.ABBYY,
            Dictionaries = new List<string>
            {
                "DictionaryName1",
                "DictionaryName2"
            },
            // provide full path to the input and output folders
            InputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input"),
            OutputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output")
        };

        lwClient.TranslateBatchFile(translateBatchFileRequest);
    }

    private static void TranslateBatchFileWithLinguisticOptions(EdgeLanguageWeaverClient lwClient)
    {
        Dictionary<string, string> linguisticOptions = new Dictionary<string, string>();
        linguisticOptions.Add("formality", "Informal");

        var translateBatchFileRequest = new EdgeTranslateBatchFileRequest()
        {
            LanguagePairId = "EngJpn_Generic_SRV_TNMV_8_5_x_6",
            LinguisticOptions = linguisticOptions,
            Dictionaries = new List<string>
            {
                "DictionaryName1",
                "DictionaryName2"
            },
            // provide full path to the input and output folders
            InputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input"),
            OutputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output"),
        };

        lwClient.TranslateBatchFile(translateBatchFileRequest);
    }
}